patroni/ha.py

```diff
@@ -372,6 +396,9 @@ def enforce_master_role(self, message, promote_message):
                 self.release_leader_key_voluntarily()
                 return 'Not promoting self because watchdog could not be activated'
 
+        if self.state_handler.get_master_timeline() > 1:
```

Comment:
I wonder if it makes sense to also erase the history stored in the DCS when we detect the timeline 1, as we don't want to carry the leftovers of a pre-pg_upgrade history over to the new cluster.